extensions-core/kafka-indexing-service/src/test/java/org/apache/druid/indexing/kafka/supervisor/KafkaSupervisorTest.java

```diff
@@ -585,6 +585,113 @@ public void testLatestOffset() throws Exception
     );
   }
 
+  /**
+   * Test if always use earliest offset on newly discovered partitions
+   */
+  @Test
+  public void testLatestOffsetOnDiscovery() throws Exception
```

Comment:
Ideally this test can be simplified to be a small extension of what's done in `testLatest` (right now it seems to be doing more work than is necessary).